File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        first, sec = nums[0], nums[1]
        if first < sec:
            first, sec = sec, first

        for i in range(2, len(nums)):
            if first < nums[i]:
                sec = first
                first = nums[i]
            elif sec < nums[i]:
                sec = nums[i]

        ans = (first - 1) * (sec - 1)
        return ans

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int first = nums[0], sec = nums[1];
        if(first < sec){
            swap(first, sec);
        }
       
        for(int i=2; i<nums.size(); i++){
            if(first < nums[i]){
                sec = max(sec, first);
                first = nums[i];
            }else if(sec < nums[i])
                sec = nums[i];
        }
        return (first-1) * (sec-1);
    }
};